gtsfm/loader/colmap_loader.py

```diff
@@ -0,0 +1,179 @@
+"""Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+Authors: John Lambert
+"""
+
+import os
+from typing import Optional
+
+from gtsam import Cal3Bundler, Pose3
+
+import gtsfm.utils.images as img_utils
+import gtsfm.utils.io as io_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class ColmapLoader(LoaderBase):
+    """Simple loader class that reads a dataset with metadata formatted in the COLMAP style.
+
+    Folder layout structure:
+    - RGB Images: images/
+    - Intrinsics data (optional): cameras.txt (see https://colmap.github.io/format.html#cameras-txt)
+    - Extrinsics data (optional): images.txt (see https://colmap.github.io/format.html#images-txt)
+
+    If explicit intrinsics are not provided, the exif data will be used.
+    """
+
+    def __init__(
+        self,
+        colmap_files_dirpath: str,
+        images_dir: str,
+        use_gt_intrinsics: bool = True,
+        use_gt_extrinsics: bool = True,
+        max_frame_lookahead: int = 1,
+        max_resolution: int = 1080
+    ) -> None:
+        """Initializes to load from a specified folder on disk.
+
+        Args:
+            colmap_files_dirpath: path to directory containing COLMAP-exported data, with images.txt
+                and cameras.txt files
+            images_dir: path to directory containing images files
+            use_gt_intrinsics: whether to use ground truth intrinsics
+            use_gt_extrinsics: whether to use ground truth extrinsics
+            max_frame_lookahead: if images were sequentially captured, maximum number
+               of consecutive frames to consider for matching/co-visibility. Defaults to 1, i.e. 
+               assuming data is not sequentially captured.
```

Comment:
I do not get this. If the frames are not in a sequence, even the value 1 doesn't make sense.